src/lib.rs

```diff
@@ -99,6 +100,23 @@
 #![deny(missing_docs)]
 
 pub use bitcoin;
+
+#[cfg(not(any(
+    any(feature = "std", feature = "no-std"),
+    all(feature = "std", feature = "no-std")
+)))]
+compile_error!("at least one of the `std` or `no-std` features must be enabled, but not both");
```

Comment:
In f8855d172471b56ef17d8f248f3c029f8895905b:

This makes me uncomfortable. Are you able to make this work such that `std` just overrides `no-std`? For that matter, are you able to eliminate the `no-std` feature and just do nostd stuff whenever `std` is not enabled?

Reply:
Grepping around I actually don't see where the `no-std` feature is even used.

Reply:
I agree that having both `std` and `no-std` features is weird. Best practice is to have a single `std` feature that's enabled by default that turns `std` usage on. The [Rust book agrees](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification). They specifically say:

>For example, if you want to optionally support [`no_std`](https://doc.rust-lang.org/reference/names/preludes.html#the-no_std-attribute) environments, **do not** use a `no_std` feature. Instead, use a `std` feature that *enables* `std`.

The main issue is that the only way to enable `rust-bitcoin`'s `no_std` mode, is to enable and re-export its `no-std` feature.

I would also rather have a single `std` feature. This is why I never reference `no-std` outside of the Cargo config and `src/lib.rs`. My hope is that eventually `rust-bitcoin` will adopt best practice and remove the `no-std` feature.

---
This is somewhat related to rust-bitcoin/rust-bitcoin#758.

Reply:
Ah, I see, I didn't realize that rust-bitcoin was similarly broken. Let's see if we can fix that now that we have a msrv bump..

Reply:
What if you removed this line and just set `no_std = ["bitcoin/no_std"]` in Cargo.toml?

I agree that it's ugly but at least it won't break compilation when users enable features.

Reply:
Removed the compiler error.